### PR TITLE
Trharris/surfacehubauth

### PR DESCRIPTION
**Pull request by TrevorJoelHarris**
For more information about how to contribute to this repo, visit [this page](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md).

## Description

Added surfaceHub to the list of hostPlatforms where teamsjs will ask the host to handle authentication. Updated, added, and fixed unit test bugs in the authentication unit tests.

### Main changes in the PR:

1. Added surfaceHub to the list of hostPlatforms where teamsjs will ask the host to handle authentication
2. Did some work to make the authentication unit tests *more* atomic. More can be done, but now we have increased the number of unit tests and have fixed at least one instance of unit tests interfering with each other.

### Validation performed:

1. Ran all unit tests

### Unit Tests added:

Yes

### End-to-end tests added:

No

### Change file added:

Yes

### CODE REVIEW

**Review by AE-MS — APPROVED** [review]
:shipit:

**Review by AE-MS — APPROVED** [review]
:shipit:

**Review by shrshindeMSFT — APPROVED** [review]

File: packages/teams-js/test/public/authentication.spec.ts [hunk 1]
```diff
@@ -51,12 +51,14 @@ describe('Testing authentication capability', () => {
     HostClientType.desktop,
     HostClientType.android,
     HostClientType.ios,
+    HostClientType.ipados,
```
Comment (lakhveerk) [inline packages/teams-js/test/public/authentication.spec.ts:54]: why are we adding this extra ipados?

Comment (lakhveerk) [reply packages/teams-js/test/public/authentication.spec.ts:54]: Please don't be blocked on my comment (I am logging off for the day). I was asking just out of curiosity.

Comment (TrevorJoelHarris) [reply packages/teams-js/test/public/authentication.spec.ts:54]: It got added to the authenticate function a few months ago but this unit test didn't get updated at the time.

**Pull request merged**